src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-heal-network.ts

```diff
@@ -0,0 +1,311 @@
+import { css, CSSResultGroup, html, LitElement, TemplateResult } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { fireEvent } from "../../../../../common/dom/fire_event";
+import "../../../../../components/ha-circular-progress";
+import { createCloseHeading } from "../../../../../components/ha-dialog";
+import {
+  fetchNetworkStatus,
+  healNetwork,
+  stopHealNetwork,
+  subscribeHealNetworkProgress,
+  ZWaveJSNetwork,
+} from "../../../../../data/zwave_js";
+import { haStyleDialog } from "../../../../../resources/styles";
+import { HomeAssistant } from "../../../../../types";
+import { ZWaveJSHealNetworkDialogParams } from "./show-dialog-zwave_js-heal-network";
+
+@customElement("dialog-zwave_js-heal-network")
+class DialogZWaveJSHealNetwork extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private entry_id?: string;
+
+  @state() private _network?: ZWaveJSNetwork;
+
+  @state() private _status?: string;
+
+  @state() private _heal_node_status?: { [key: number]: string };
+
+  @state() private _progress_total = 0;
+
+  @state() private _progress_finished = 0;
+
+  @state() private _progress_in_progress = 0;
+
+  private _subscribed?: Promise<UnsubscribeFunc>;
+
+  public async showDialog(
+    params: ZWaveJSHealNetworkDialogParams
+  ): Promise<void> {
+    this._heal_node_status = undefined;
+    this.entry_id = params.entry_id;
+    this._fetchData();
+  }
+
+  protected render(): TemplateResult {
+    if (!this.entry_id) {
+      return html``;
+    }
+
+    return html`
+      <ha-dialog
+        open
+        @closed=${this.closeDialog}
+        .heading=${createCloseHeading(
+          this.hass,
+          this.hass.localize("ui.panel.config.zwave_js.heal_network.title")
+        )}
+      >
+        ${!this._status
+          ? html`
+              <div class="flex-container">
+                <ha-svg-icon
+                  .path=${mdiStethoscope}
+                  class="introduction"
+                ></ha-svg-icon>
+                <div class="status">
+                  <p>
+                    ${this.hass.localize(
+                      "ui.panel.config.zwave_js.heal_network.introduction"
+                    )}
+                  </p>
+                </div>
+              </div>
+              <p>
+                <em>
+                  ${this.hass.localize(
+                    "ui.panel.config.zwave_js.heal_network.traffic_warning"
+                  )}
+                </em>
+              </p>
+              <mwc-button slot="primaryAction" @click=${this._startHeal}>
+                ${this.hass.localize(
+                  "ui.panel.config.zwave_js.heal_network.start_heal"
+                )}
+              </mwc-button>
+            `
+          : ``}
+        ${this._status === "started"
+          ? html`
+              <div class="flex-container">
+                <ha-circular-progress active></ha-circular-progress>
```

Comment:
If we have a `linear-progress`, we dont also need a `circular-progress` I think?